aiohttp/connector.py

```diff
@@ -850,6 +850,39 @@ async def _resolve_host(
                     await trace.send_dns_cache_hit(host)
             return result
 
+        #
+        # If multiple connectors are resolving the same host, we wait
+        # for the first one to resolve and then use the result for all of them.
+        # We use a throttle event to ensure that we only resolve the host once
+        # and then use the result for all the waiters.
+        #
+        # In this case we need to create a task to ensure that we can shield
+        # the task from cancellation as cancelling this lookup should not cancel
+        # the underlying lookup or else the cancel event will get broadcast to
+        # all the waiters across all connections.
+        #
+        host_resolved = asyncio.create_task(
+            self._resolve_host_with_throttle(key, host, port, traces)
+        )
+        try:
+            return await asyncio.shield(host_resolved)
+        except asyncio.CancelledError:
+
+            def drop_exception(fut: "asyncio.Future[List[Dict[str, Any]]]") -> None:
```

Comment:
No need to quote the annotation since there's no py310-only syntax here:
```suggestion
            def drop_exception(fut: asyncio.Future[List[Dict[str, Any]]]) -> None:
```

Reply:
I'm not sure asyncio.Future is sub-scriptable on older python.  Will drop it and see if the CI complains

Reply:
Just tried, it's not.

Reply:
restored quoting 38e0649023c32bcb2c986a0aa744f5ad17799c32

Reply:
Ah, I didn't realize that.